ast/ast.go

```diff
@@ -732,8 +732,12 @@ func (n *MappingValueNode) String() string {
 		return fmt.Sprintf("%s%s:\n%s", space, n.Key.String(), n.Value.String())
 	} else if m, ok := n.Value.(*MappingNode); ok && m.IsFlowStyle {
 		return fmt.Sprintf("%s%s: %s", space, n.Key.String(), n.Value.String())
+	} else if m, ok := n.Value.(*MappingNode); ok && len(m.Values) == 0 {
```

Comment:
Could you combine with the above condition ? 

```go
else if m, ok := n.Value.(*MappingNode); ok && (m.IsFlowStyle || len(m.Values) == 0) {
}
```

Reply:
Fixed at https://github.com/goccy/go-yaml/pull/98/commits/f92ff9457879ab953fb09a8a68ee7d42276d007b